Bottom: 8454fc5e0e44c278f6a8a76c326c143097e4e3f2
Top:    4f8334771aa3c4be851c0624ca1b2c7d2feeded9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-01 13:04:15 +0100

Temporary debug for toporouter - switch back to cmil Coord units


---

diff --git a/src/const.h b/src/const.h
index c82e90a..f554fd4 100644
--- a/src/const.h
+++ b/src/const.h
@@ -68,10 +68,10 @@
 #define LN_2_OVER_2		0.346573590
 
 /* PCB/physical unit conversions */
-#define COORD_TO_MIL(n)	((n) / 25400.0)
-#define MIL_TO_COORD(n)	((n) * 25400.0)
-#define COORD_TO_MM(n)	((n) / 1000000.0)
-#define MM_TO_COORD(n)	((n) * 1000000.0)
+#define COORD_TO_MIL(n)	((n) / 100.0)
+#define MIL_TO_COORD(n)	((n) * 100.0)
+#define COORD_TO_MM(n)	((n) * 0.000254)
+#define MM_TO_COORD(n)	((n) / 0.000254)
 #define COORD_TO_INCH(n)	(COORD_TO_MIL(n) / 1000.0)
 #define INCH_TO_COORD(n)	(MIL_TO_COORD(n) * 1000.0)
 
diff --git a/src/toporouter.c b/src/toporouter.c
index 48bb306..b5aeb1c 100644
--- a/src/toporouter.c
+++ b/src/toporouter.c
@@ -232,7 +232,7 @@ toporouter_arc_class(void)
   return klass;
 }
 
-#define MARGIN 10.0f
+#define MARGIN 10.0f /* XXX: Not sure - seems to be used to offset cairo drawing - thus could be in Coords */
 
 drawing_context_t *
 toporouter_output_init(int w, int h, char *filename) 
@@ -1156,7 +1156,7 @@ wind(toporouter_spoint_t *p1, toporouter_spoint_t *p2, toporouter_spoint_t *p3)
   dx1 = p2->x - p1->x; dy1 = p2->y - p1->y;
   dx2 = p3->x - p2->x; dy2 = p3->y - p2->y;
   rval = (dx1*dy2)-(dy1*dx2);
-  return (rval > 0.0001) ? 1 : ((rval < -0.0001) ? -1 : 0);
+  return (rval > 0.0001) ? 1 : ((rval < -0.0001) ? -1 : 0); /* XXX: Depends on PCB coordinate scaling */
 }
 
 static inline void
